### ISSUE TITLE
How to prepare the lable file?

### ISSUE THREAD

**Issue by VCBE123** [wulalago/UNet#5]
In order to visualize the mask, the pixes value from 0 to 255. And the dataset contains 3 classes.
The background is 0, the class one is 128, class two is 255.
I use sigmoid function and BCELoss.
In the dataloader, do i need one-hot encode the lable or not?
Can i just load the original mask file for loss calculation and evaluation?

**Comment by VCBE123** [comment]
I find the code. Sorry to interrupt.

**Comment by VCBE123** [comment]
For the multi-class segmentation. How to choose the output activation layer?
Sigmoid or softmax  or log_softmax?

**Comment by wulalago** [comment]
softmax or log_softmax.  If you use the BCELoss function and the activation layer of output is log_softmax, it is equal with BCEWithLogitsLoss without activation layer of output. Or you can still use the softmax, anyway, I think it is no difference between these two method and is fine to keep the training stage and test stage equally whatever softmax or log_softmax.

**Comment by VCBE123** [comment]
Thank you for your advice.  Finaly I use softmax and one-hot encoding and BCELoss.
